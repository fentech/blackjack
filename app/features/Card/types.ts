export type Suits = "hearts" | "diamonds" | "spades" | "clubs";
export type Ranks =
  | 2
  | 3
  | 4
  | 5
  | 6
  | 7
  | 8
  | 9
  | 10
  | "jack"
  | "queen"
  | "king"
  | "ace";
