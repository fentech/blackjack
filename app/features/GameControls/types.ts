export type Person = "player" | "dealer";
export type Turn = Person | null;
