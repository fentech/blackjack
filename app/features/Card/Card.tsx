import React from "react";
import styled from "styled-components/native";

export interface CardProps {
  suit: "hearts" | "diamonds" | "spades" | "clubs";
  rank: 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | "jack" | "queen" | "king" | "ace";
}

const CardText = styled.Text`
  font-size: 18px;
`;

const Card: React.FC<CardProps> = ({ suit, rank }) => {
  return (
    <CardText>
      {rank} of {suit}
    </CardText>
  );
};

export default Card;
