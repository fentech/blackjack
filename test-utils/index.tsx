import React from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { ApplicationProvider } from "@ui-kitten/components";
import { mapping, light as lightTheme } from "@eva-design/eva";
import store from "../app/store";

export const WithRedux: React.FC = ({ children }) => {
  return (
    <ApplicationProvider mapping={mapping} theme={lightTheme}>
      <Provider store={store}>{children}</Provider>
    </ApplicationProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options: Omit<RenderOptions, "queries"> = {}
) => render(ui, { wrapper: WithRedux, ...options });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
