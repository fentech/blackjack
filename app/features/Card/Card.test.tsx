import React from "react";
import { render } from "test-utils";
import Card from "./Card";
import { Ranks, Suits } from "./types";

describe("Card", () => {
  it("matches snapshot", (): void => {
    const suits: Suits[] = ["clubs", "hearts", "clubs", "spades"];
    const ranks: Ranks[] = ["nine", "ace", "two", "jack"];

    suits.forEach((suit, i) => {
      const rank = ranks[i];
      const { baseElement } = render(<Card suit={suit} rank={rank} />);

      expect(baseElement).toMatchSnapshot();
    });
  });
});
