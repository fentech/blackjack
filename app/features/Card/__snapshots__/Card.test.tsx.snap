// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#fff",
              "borderRadius": 3,
              "height": 115,
              "justifyContent": "center",
              "marginRight": 10,
              "paddingBottom": 4,
              "paddingLeft": 4,
              "paddingRight": 4,
              "paddingTop": 4,
              "width": 80,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/features/Card/cards/9_of_clubs.png",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Card matches snapshot 2`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#fff",
              "borderRadius": 3,
              "height": 115,
              "justifyContent": "center",
              "marginRight": 10,
              "paddingBottom": 4,
              "paddingLeft": 4,
              "paddingRight": 4,
              "paddingTop": 4,
              "width": 80,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/features/Card/cards/ace_of_hearts.png",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Card matches snapshot 3`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#fff",
              "borderRadius": 3,
              "height": 115,
              "justifyContent": "center",
              "marginRight": 10,
              "paddingBottom": 4,
              "paddingLeft": 4,
              "paddingRight": 4,
              "paddingTop": 4,
              "width": 80,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/features/Card/cards/2_of_clubs.png",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Card matches snapshot 4`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#fff",
              "borderRadius": 3,
              "height": 115,
              "justifyContent": "center",
              "marginRight": 10,
              "paddingBottom": 4,
              "paddingLeft": 4,
              "paddingRight": 4,
              "paddingTop": 4,
              "width": 80,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/features/Card/cards/jack_of_spades.png",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
